Add a few fields useful for DHCP Network and Range creation

ExaneServerTeam opened the pull request:
Network
-> zone_assiociations
IPRange
-> server_association_type
-> failover_association

One use case it to map IPRange to failover cluster with:
server_association_type='FAILOVER'
failover_association='YOUR_FAILOVER_CLUSTER_NAME'

yuewko commented:
LGTM.

For your considerations...

I believe `_return_fields` only specifies the *default* set of return fields. Additional return fields, can be specified (and existing return fields can be overridden) when doing a get/search, as long as they are specified in `_fields`.

(please verify the above as I am a bit out of touch)

Assuming the above is true, then I feel we should only include *commonly* used fields in `_return_fields`.

ExaneServerTeam commented:
@yuewko Thanks for this update. I indeed did not review the _search in the InfobloxObject class.
You are indeed able to overide return_fields function parameters. Would be very helpful for my implementation.
I still think that the member association is a basic setting of an IPRange (That needs review for conformance)

yuewko commented:
@ExaneServerTeam 

> I still think that the member association is a basic setting of an IPRange

I'm not sure I agree that it is a *basic* setting of an IPRange.
Having said that, I'm afraid I'm a bit out of touch with this - I'll let the core team make the call.

adsri commented on infoblox_client/objects.py:
```diff
@@ -731,13 +732,15 @@ class DhcpOption(SubObjects):
     _fields = ['start_addr', 'end_addr', 'network_view',
-               'network', 'extattrs', 'disable']
+               'network', 'extattrs', 'disable', 'server_association_type',
+               'failover_association']
```
According to the wapi document,
The basic version of the object contains the field(s): comment, end_addr, network, network_view, start_addr. 
Hence, i feel we should use _return_field override instead of making these fields as basic fields to the class.

yuewko replied:
@adsri , please see my comment below (https://github.com/infobloxopen/infoblox-client/pull/187#discussion_r312081867)

adsri commented on infoblox_client/objects.py:
```diff
@@ -448,12 +448,13 @@ def get_v6_class(cls):
     _fields = ['network_view', 'network', 'template',
-               'options', 'members', 'extattrs', 'comment']
+               'options', 'members', 'extattrs', 'comment',
+               'zone_associations']
```
the basic version of network object donot include zone_associations field. hence, it should not be included in Network class as basic fields

yuewko replied:
@adsri , I think there's a misunderstanding here.

What I was trying to say in my comment above (https://github.com/infobloxopen/infoblox-client/pull/187#issuecomment-369620183) is that, non-basic attributes should not be included in `_return_fields`.

We must, however, still include those non-basic attributes in `_fields` - otherwise, you cannot specify those attributes (to override default `_return_fields`) AT ALL when doing get/search.

Hope I make my clear this time.

saiprasannasastry commented:
@yuewko none of the new values added in the _return_fields are part of basic fields , so @ExaneServerTeam  member association is not a part of base object, so if you undo the changes for _return_fields, we can merge this

ExaneServerTeam commented:
@yuewko Thanks for the followup.
I rebased this little change on current master and cleaned up anything related to IP range.
Just one field is appended (Then one I am interrested in).
It should be ready for a merge.

saiprasannasastry commented:
LGTM
